### PR TITLE
Rework publish scripts [non-fork pr]

### PR DESCRIPTION
**Pull request by frant-hartm**
The goal is to be able to publish packages for both OS and EE and for any versions ve release (stable, snapshot, development releases and betas).

I have moved all the code to prepare the package to `build-[deb|rpm|homebrew].sh` scripts. These are then called from GH action `publish-packages.yaml`

The scrips use the following env variables as input:

HZ_DISTRIBUTION - `hazelcast` or `hazelcast-enterprise` for OS or EE version of the distribution package
HZ_VERSION - version of hazelcast
PACKAGE_VERSION - version of the package based on HZ_VERSION
PUBLISH - if to upload final package to the repository

We can't access secrets from a fork. For this reason, all PRs which need to run the automation will have to be created from this repository's branches.

HZ-536
HZ-533
HZ-508

TODOs:

- [x] move HZ_VERSION to PACKAGE_VERSION conversion to common.sh
- [x] update readme
- [x] add some documentation (started here https://docs.google.com/document/d/1fIC7hzZDMXg7PN_Kvyxg1r3kQuBqvLjG-kVNJPy1uyQ/edit?usp=sharing)
- [x] change to production repositories when all is done and testing finished
- [ ] upgrade on homebrew alias doesn't seem to work
- [ ] homebrew `conflicts_with` doesn't work as I expect
- [x] should we make this repository private? maybe rename?

### CODE REVIEW

File: common.sh [hunk 1]
```diff
@@ -23,9 +23,8 @@ if [ "${EVENT_NAME}" == "pull_request" ]; then
   export DEBIAN_REPO_BASE_URL="https://repository.hazelcast.com/debian-test-local"
   export RPM_REPO_BASE_URL="https://repository.hazelcast.com/rpm-test-local"
 
-  # TODO change when the access is setup
-  export BREW_GIT_REPO_NAME="frant-hartm/homebrew-hz"
-  export BREW_REPO_NAME="frant-hartm/hz"
+  export BREW_GIT_REPO_NAME="hazelcast/homebrew-hz-test"
+  export BREW_REPO_NAME="hazelcast/hz-test"
```
Comment (ldziedziul) [inline common.sh:27]: What do you think about deduplication with small bash magic
```suggestion
  export BREW_REPO_NAME="hazelcast/hz-test"
  export BREW_GIT_REPO_NAME="${BREW_REPO_NAME//\///homebrew-}"
```

**Review by ldziedziul — APPROVED** [review]

File: common.sh (hunk 1 above)
Comment (ldziedziul) [reply common.sh:27]: Also, maybe rename `BREW_REPO_NAME` to `BREW_TAP_NAME` ?

File: README.md [hunk 2]
```diff
@@ -1,207 +1,143 @@
...
+## Running Hazelcast
 
-Please use the following command to run [SpotBugs](https://spotbugs.github.io/) analysis:
+After successful installation all commands from Hazelcast distribution
+`bin` directory should be on path.
 
-```
-mvn -P spotbugs clean compile
-```
+Run the following command to start a Hazelcast server with the default configuration:
```
Comment (ldziedziul) [inline README.md:139]: What do you think about mentioning that additional options can be checked with `hz start --help`?

**Review by ldziedziul — APPROVED** [review]
LGTM, just a small note

File: README.md (hunk 2 above)
Comment (frant-hartm) [reply README.md:138]: Added.

**Pull request merged**